Feature/remember users after login

robcole opened the pull request:
Adds support for remembering users after logging into slack. Defaults to 1 week 
cookie expiration.

All comments appreciated!

robcole committed:
Add auth_token to User model

Uses has_secure_token to add auth_token for
"remember me" support.

robcole committed:
Update Production env to force_ssl

Forces production site to use SSL for cookie-based
login support.

robcole committed:
Add support for remembering Slack login

Adds full support for remembering users after
slack login. Defaults to 1-week expiration on
cookies.

robcole committed:
Merge branch 'develop' into feature/remember-users-after-login

# Conflicts:
#	db/schema.rb

robcole committed:
Update schema

Adds incorrectly removed aasm_state attribute.

robcole committed:
Fix rubocop warnings

fozcodes commented on spec/features/authentication_spec.rb:
```diff
@@ -14,6 +16,11 @@
       end
+
+      scenario 'changes the login text' do
+        expect(page).not_to have_content('Sign In With Slack')
```
I think we should be using i18n here. That's my bad - I should have done that when I put in that navbar item.

fozcodes commented on spec/features/authentication_spec.rb:
```diff
@@ -14,6 +16,11 @@
+
+      scenario 'changes the login text' do
+        expect(page).not_to have_content('Sign In With Slack')
+        expect(page).to have_content('Sign Out')
```
Same here - i18n

fozcodes commented on app/views/layouts/_navigation_links.html.erb:
```diff
@@ -1,3 +1,7 @@
+<% if current_user %>
+  <%= link_to "Sign Out", logout_path, class: 'navbar-brand' %>
+<% else %>
   <%= link_to 'Sign In With Slack', login_path('slack'), class: 'navbar-brand' %>
```
If we use i18n for these, we won't have to worry about changing the wording and breaking tests.

fozcodes commented on app/views/layouts/_navigation_links.html.erb:
```diff
@@ -1,3 +1,7 @@
+  <%= link_to "Sign Out", logout_path, class: 'navbar-brand' %>
+<% else %>
   <%= link_to 'Sign In With Slack', login_path('slack'), class: 'navbar-brand' %>
+<% end %>
```
Could we move this conditional into a view helper?

fozcodes commented:
Hey Rob, I think this looks good. A couple of small comments from me. Another thing I think we need to do overall is get off the Controller tests and on to Integration tests, but that's probably something we can refactor after we get this thing out into the world.

Let me know what you think!

robcole committed:
Add i18n support for Slack Login

Adds i18n support for login text. Moves conditional
for slack login link to application helper.

robcole committed:
Fix parameter alignment in application helper

Align parameters in application helper according
to Rubocop recommendations.

robcole commented:
@fosberyi - Made the recommended changes, everything made sense. I'm not sure about controller spec vs. integration specs when we're testing cookies (happy to revisit that down the road) but beyond that, here you go.

fozcodes commented:
@robcole sweet looks good man! Merge away!

robcole commented:
I don't have merge permissions - feel free to merge or let me know what to do next in the event I'm missing something here.
